source/fundamentals/linq.txt

```diff
@@ -248,6 +248,43 @@ The result of the preceding example contains the following documents:
     { "name" : "Crystal Room", "cuisine" : "Italian" }
     { "name" : "Forlinis Restaurant", "cuisine" : "Italian" }
 
+.. _csharp-linq-takewhile:
+
+Limit on a Condition
+````````````````````
+
+You can use the ``TakeWhile()`` LINQ method in a ``Select()`` projection
+to return array field elements while a specified condition is true, then
+skip the remaining elements.
+
+This example uses the following ``Student`` class to model documents
+that contain an array field ``Grades``:
+
+.. code-block:: csharp
+
+   public class Student
+   {
+       public string Name { get; set; }
+       public int[] Grades { get; set; }
+   }
+
+The following code shows how to use the ``TakeWhile()`` method to return
+any ``Grades`` array elements that are greater than ``90`` and skip the
+rest of the array:
+
+.. code-block:: csharp
+
+   var query = queryableCollection
+       .Select(s => s.Grades.TakeWhile(g => g > 90);
+
+The results resemble the following documents:
+
+.. code-block:: json
+   :copyable: false
+
+   { "name" : "Maura Day", "grades" : [92, 97] }
+   { "name" : "Saul Curzon", "grades" : [100, 95, 91] }
```

Comment:
The results do NOT resemble these documents.

The result is an array of `Grades`.  It might better to say the results are:

    [92, 97]
    [100, 95, 91]

Though I have no idea if the values are correct because I don't know the test data.

Reply:
Thanks! I am going to test this example now that the driver is released. I must have interpreted the test from the source code incorrectly.

Reply:
Fixed code and results